validation/src/main/java/io/serverlessworkflow/validation/WorkflowValidatorImpl.java

```diff
@@ -27,24 +28,24 @@
 import io.serverlessworkflow.api.states.*;
 import io.serverlessworkflow.api.switchconditions.DataCondition;
 import io.serverlessworkflow.api.switchconditions.EventCondition;
+import io.serverlessworkflow.api.utils.Utils;
 import io.serverlessworkflow.api.validation.ValidationError;
 import io.serverlessworkflow.api.validation.WorkflowSchemaLoader;
-import java.util.ArrayList;
+import java.io.IOException;
+import java.util.Collection;
 import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
-import org.everit.json.schema.Schema;
-import org.everit.json.schema.ValidationException;
-import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class WorkflowValidatorImpl implements WorkflowValidator {
 
   private static final Logger logger = LoggerFactory.getLogger(WorkflowValidatorImpl.class);
   private boolean schemaValidationEnabled = true;
-  private List<ValidationError> validationErrors = new ArrayList<>();
-  private Schema workflowSchema = WorkflowSchemaLoader.getWorkflowSchema();
+  private Collection<ValidationError> validationErrors = new LinkedHashSet<>();
```

Comment:
List are always ordered, but  I think the requirement here is that errors are sorted by insertion order and this is also provided by LinkedHashSet (and not necessarily by List, because implementations can always set the position of indexes in a different way than insertion order)  
The reason I changed to Collection is because it allows using a Set implementation, which prevents duplication, which is something that I think should also be a requirement for validations errors. To prevent duplication, I added hashcode and equals methods to ValidationError class.  
I think the discussion should be if we change to Collection or directly to Set, because declaring validation errors as Set ensures there is not duplication (and we LinkedHashSet you can keep the previous requirement of having errors sorted by order of appearance)  
To stress my poin, I want to note that the validation library API returns a set with the validation errors.

Reply:
ok yeah im not sure this duplication requirement is valid one but will think more about it (per comment above also please let me know if we have a test for this possible duplication)
either way i think that having iteration order is as much important as possibly this duplication issue if that is indeed an issue

Reply:
agree. i think having a test that generates same custom validation error on 2+ states and then see if this uniqueness kicks in and does not remove the error that actually is valid (so not duplicate but same for a different wf state) would be nice